pkg/sharing/files.go

```diff
@@ -3,36 +3,77 @@ package sharing
 import (
 	"github.com/cozy/cozy-stack/pkg/consts"
 	"github.com/cozy/cozy-stack/pkg/couchdb"
+	"github.com/cozy/cozy-stack/pkg/crypto"
 	"github.com/cozy/cozy-stack/pkg/instance"
 	"github.com/cozy/cozy-stack/pkg/vfs"
 )
 
+// MakeXorKey generates a key for transforming the file identifiers
+func MakeXorKey() []byte {
+	random := crypto.GenerateRandomBytes(8)
+	result := make([]byte, 2*len(random))
+	for i, val := range random {
+		result[2*i] = val & 0xf
+		result[2*i+1] = val >> 4
+	}
+	return result
+}
+
+// XorID transforms the identifier of a file to a new identifier, in a
+// reversible way: it makes a XOR on the hexadecimal characters
+func XorID(id string, key []byte) string {
+	l := len(key)
+	buf := []byte(id)
+	for i, c := range buf {
+		switch {
+		case '0' <= c && c <= '9':
+			c = (c - '0') ^ key[i%l]
+		case 'a' <= c && c <= 'f':
+			c = (c - 'a' + 10) ^ key[i%l]
+		case 'A' <= c && c <= 'F':
+			c = (c - 'A' + 10) ^ key[i%l]
+		default:
+			continue
+		}
+		if c < 10 {
+			buf[i] = c + '0'
+		} else {
+			buf[i] = (c - 10) + 'a'
+		}
+	}
+	return string(buf)
+}
+
 // EnsureSharedWithMeDir returns the shared-with-me directory, and create it if
 // it doesn't exist
-func EnsureSharedWithMeDir(inst *instance.Instance) error {
+func EnsureSharedWithMeDir(inst *instance.Instance) (*vfs.DirDoc, error) {
 	fs := inst.VFS()
 	dir, _, err := fs.DirOrFileByID(consts.SharedWithMeDirID)
 	if err != nil {
-		return err
+		return nil, err
```

Comment:
A missing sharedWithMe dir returns an `err`, you might add:
`if !couchdb.IsNotFoundError(err) {
  return nil, err
}`